### PR TITLE
Support broadcast add & mul on CUDA (fixed)

### PR DESCRIPTION
**Pull request by li-plus**
Should fix #2191. Missed `ne11` term for `ky` in the previous version. Now fixed and generation is correct.

### CODE REVIEW

File: ggml-cuda.cu [hunk 1]
```diff
@@ -2308,19 +2306,12 @@ inline void ggml_cuda_op_mul(
...
-        const int64_t i11 = i1*ne11 + i01%ne11; // broadcast src1 across src0
-
-        float * src0_ddf_i01 = src0_ddf_i + i01*ne00;
-        float * src1_ddf_i01 = src1_ddf_i + i11*ne10;
-        float * dst_ddf_i01  = dst_ddf_i + i01*ne00;
-
-        // compute
-        mul_f32_cuda(src0_ddf_i01, src1_ddf_i01, dst_ddf_i01, ne00, ne10, cudaStream_main);
-    }
+    mul_f32_cuda(src0_ddf_i, src1_ddf_i, dst_ddf_i, ne00*i01_diff, ne10*ne11, cudaStream_main);
```
Comment (JohannesGaessler) [inline ggml-cuda.cu:2314]: The broadcasting logic here is still incorrect. The implementation on master broadcasts the values **per row** while this broadcasts the values after flattening both tensors. As long as `ne11 == 1` this doesn't make a difference but I don't think that this is the implementation that we should be using.

**Review by JohannesGaessler — CHANGES_REQUESTED** [review]

File: ggml-cuda.cu (hunk 1 above)
Comment (li-plus) [reply ggml-cuda.cu:2314]: I noticed that the column number of `src0` and `src1` is ensured to be the same by checking `ggml_can_repeat_rows`, so broadcasting after flattening 2d sub blocks should be the same as broadcasting every row. Did I miss something?

https://github.com/ggerganov/llama.cpp/blob/32c54116318929c90fd7ae814cf9b5232cd44c36/ggml.c#L5228-L5235

Comment (JohannesGaessler) [reply ggml-cuda.cu:2314]: Okay, according to the commit history the following seems to have happened: I implemented broadcasting for multiplication both in dimension 0 and dimension 1. @ggerganov then added the additional requirement that dimension 0 must be equal via `ggml_can_repeat_rows` which effectively limits broadcasting to dimension 1. If that is indeed the specification to which broadcasting should be implemented then the broadcasting logic in this PR is correct.

Comment (ggerganov) [reply ggml-cuda.cu:2314]: For now let's broadcast only over dimension 1 - later we'll fix the TODOs and support dimension 0 broadcasts.
Can also obsolete `ggml_scale()` when we do that.

**Review by ggerganov — APPROVED** [review]

**Pull request merged**